app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -636,23 +638,44 @@ export default Ember.Component.extend({
     return null;
   },
 
+  @computed
+  pastePlaceholder() {
+    return I18n.t('pasting');
+  },
+
   paste(e) {
     const clipboard = clipboardData(e);
     const types = clipboard.types;
-    let preventDefault = false;
+    const plainText = clipboard.getData("text/plain");
+    const placeholder = plainText || this.get('pastePlaceholder');
+    let handled = false;
 
-    if (types.some(t => t === "text/plain")) {
-      const text = clipboard.getData("text/plain");
-      const table = this._detectTable(text);
+    if (plainText) {
+      const table = this._detectTable(plainText);
```

Comment:
minor complaint but actually here we are not detecting a table, but trying to extract one, so the name could be better (you are actually detecting a table in the middle of detectTable logic).

```
cons table = this._extractTable(plainText);
```

would make more sense I think

Reply:
btw, are we sure `clipboard.types` will always return an array ?

Reply:
yes.